.evergreen/download-mongodb.sh

```diff
@@ -388,10 +420,18 @@ get_mongodb_download_url_for ()
              MONGODB_50="http://downloads.10gen.com/linux/mongodb-linux-x86_64-enterprise-debian11${DEBUG}-${VERSION_50}.tgz"
              # SERVER-62299 Added support for Debian 11 in 5.0.8 and 6.0.0-rc0
       ;;
+      linux-debian-12*)
```

Comment:
Note that debian is not sorted in descending order but has too many versions for us to fix in this PR 😅